src/vos/vos_io.c

```diff
@@ -1010,7 +1019,14 @@ stop_on_empty(struct vos_io_context *ioc, uint64_t cond, int *rc)
 		return true;
 	}
 
-	if (ioc->ic_ts_set->ts_flags & cond)
+	if (iod != NULL && ioc->ic_ts_set->ts_flags & VOS_OF_COND_PER_AKEY) {
+		/** Per akey flags have been specified */
+		flags = iod->iod_flags;
```

Comment:
If the application set "iod_flags" with VOS_OF_COND_PER_AKEY, then the check at line 1029 may pass unexpectedly. So filterout the VOS_OF_COND_PER_AKEY?
flags = iod->iod_flags & ~VOS_OF_COND_PER_AKEY;

Reply:
When doing the check for akey, we pass only VOS_OF_COND_AKEY_FETCH so it won't happen.  Otherwise, we don't pass iod.